File: commons/src/main/java/com/eirsteir/coffeewithme/commons/security/UserDetailsImpl.java
package com.eirsteir.coffeewithme.commons.security;

import java.util.Collection;
import lombok.*;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

@ToString
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserDetailsImpl implements UserDetails {

  private Long id;
  private String email;
  private String nickname;
  private String password;
  Collection<? extends GrantedAuthority> roles;

  @Override
  public Collection<? extends GrantedAuthority> getAuthorities() {
    return roles;
  }

  @Override
  public String getPassword() {
    return password;
  }

  @Override
  public String getUsername() {
    return nickname;
  }

  public String getNickname() {
    return nickname;
  }

  @Override
  public boolean isAccountNonExpired() {
    return true;
  }

  @Override
  public boolean isAccountNonLocked() {
    return true;
  }

  @Override
  public boolean isCredentialsNonExpired() {
    return true;
  }

  @Override
  public boolean isEnabled() {
    return true;
  }
}

File: commons/src/main/kotlin/com/eirsteir/coffeewithme/commons/security/UserDetailsImpl.kt
package com.eirsteir.coffeewithme.commons.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails


data class UserDetailsImpl(
    val id: Long,
    val email: String,
    val nickname: String,
    private val password: String,
    var roles: Collection<GrantedAuthority>,
    private val isAccountNonExpired: Boolean = true,
    private val isAccountNonLocked: Boolean = false,
    private val isCredentialsNonExpired: Boolean = false,
    private val isEnabled: Boolean = false,
) : UserDetails {

    override fun getAuthorities() = roles

    override fun getPassword() = password

    override fun getUsername() = nickname

    override fun isAccountNonExpired() = isAccountNonExpired

    override fun isAccountNonLocked() = isAccountNonLocked

    override fun isCredentialsNonExpired() = isCredentialsNonExpired

    override fun isEnabled() = isEnabled
}